DevVehicle35-Motors/App/VanInteraction.cs

```diff
@@ -0,0 +1,134 @@
+            Console.WriteLine(string.Empty);
+            Console.WriteLine("Hi! We will help you to find the Van of your dreams...");
+            BathroomChoise();
+            opt = Console.ReadLine();
+            optN = ValidateInput(opt, 1, 2, 1);
+            switch (optN)
+            {
+                case 1: van.SetBathroom(true);
+                    break;
+                case 2: van.SetBathroom(false);
+                    break;
+                default: Console.WriteLine("Thanks for visiting us!"); Environment.Exit(0); break;
+            }
+            Console.WriteLine(string.Empty);
+            CapacityChoise();
+            opt = Console.ReadLine();
+            optN = ValidateInput(opt, 1, 3, 2);
+            if (optN == 0)
+            {
+                Console.WriteLine("Thanks for visiting us!");
+                Environment.Exit(0);
+            }
+            van.SetCapacity(optN);
+            Vcapacity = optN;
+            SizeOfBedChoise();
+            opt = Console.ReadLine();
+            optN = Vcapacity>1? ValidateInput(opt, 2, 3, 3): ValidateInput(opt, 1, 3, 3);
+            switch (optN)
+            {
+                case 1: van.SetSizeofBed(Enum.SizeOfBedsEnum.Simple);
+                    break;
+                case 2: van.SetSizeofBed(Enum.SizeOfBedsEnum.Double);
+                    break;
+                case 3: van.SetSizeofBed(Enum.SizeOfBedsEnum.KingSize);
+                    break;
+                default: break;
+            }
+            ShowDescription(van);
+        }
+
+        private static int ValidateInput(string? resp, int min, int max, int menu)
+        {
+            int opt = 0;
+            while (opt == 0)
+            {
+                try
+                {
+                    if (resp == "q")
+                    {
+                        return 0;
+                    }
+
+                    opt = int.Parse(resp);
+                    if (opt >= min && opt <= max)
+                    {
+                        return opt;
+                    }
+                    else
+                    {
+                        throw new Exception();
+                    }
+                }
+                catch (Exception e)
+                {
+                    opt = 0;
+                    Console.WriteLine("You have entered a wrong input value, try again.");
+                    switch (menu)
+                    {
+                        case 1: BathroomChoise(); break;
+                        case 2: CapacityChoise(); break;
+                        case 3: SizeOfBedChoise(); break;
+                        default: break;
+                    }
+                    resp = Console.ReadLine();
+                    Console.WriteLine("");
+                }
+            }
+            return opt;
+        }
+
+        private static void SizeOfBedChoise()
+        {
+            Console.WriteLine("Tell us, What is the size of bed of your preference?:");
```

Comment:
Think here de message is somehow confusing, think the question you wanted to ask was: What is your preferred bed size?

Reply:
Thanks for the help! The suggestion has been implemented.